ibm/resource_ibm_private_dns_permitted_network.go

```diff
@@ -0,0 +1,154 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	//"fmt"
+
+	"github.com/IBM/dns-svcs-go-sdk/dnssvcsv1"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	iserrors "github.ibm.com/Bluemix/riaas-go-client/errors"
+)
+
+const (
+	pdnsZoneID = "zone_id"
+	pdnsVpcCRN = "vpc_crn"
+)
+
+func resourceIBMPrivateDNSPermittedNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDnsPermittedNetworkCreate,
+		Read:     resourceIBMPrivateDnsPermittedNetworkRead,
+		Update:   resourceIBMPrivateDnsPermittedNetworkUpdate,
+		Delete:   resourceIBMPrivateDnsPermittedNetworkDelete,
+		Exists:   resourceIBMPrivateDnsPermittedNetworkExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsVpcCRN: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+		},
+	}
```

Comment:
WE have type parameter also for https://cloud.ibm.com/apidocs/dns-svcs#create-a-permitted-network

Add a attribute to set the permitted_network_id (the unique id for this resource)